automation/run-tests.sh

```diff
@@ -227,8 +227,16 @@ function modprobe_ovs {
     lsmod | grep -q ^openvswitch || modprobe openvswitch || { echo 1>&2 "Please run 'modprobe openvswitch' as root"; exit 1; }
 }
 
+function vdsm_tests {
+  git clone https://gerrit.ovirt.org/vdsm
+  cd vdsm
+  ./tests/network/functional/run-tests.sh --nmstate-source="$PROJECT_PATH"
+  cd -
+  rm -rf vdsm
```

Comment:
I am not sure what we can do about the vdsm clone. Any idea?

Reply:
What exactly is the problem that you see here? Cloning with git in general or the possible problems about how it is implemented here?

Reply:
The possible solution how to remove any leftover. There should be logic that will remove it no matter if the test fail or not.

Reply:
Here is my suggestion:

1) Always create a temp directory for the tests:
```bash
NMSTATE_TEMPDIR=$(mktemp -d /tmp/nmstate-test-XXXX)
```

2) Update run_exit to always remove it

3) Clone vdsm into the temp dir and run the test from there if requested
```bash
git clone -C "${NMSTATE_TEMPDIR}" --depth 1 https://gerrit.ovirt.org/vdsm
"${NMSTATE_TEMPDIR}/vdsm/tests/network/functional/run-tests.sh" --nmstate-source="$PROJECT_PATH"
```

What do you think?

Reply:
Sounds great, thanks